django/db/models/query.py

```diff
@@ -494,6 +494,10 @@ def _extract_model_params(self, defaults, **kwargs):
             if f.attname in lookup:
                 lookup[f.name] = lookup.pop(f.attname)
         params = {k: v for k, v in kwargs.items() if LOOKUP_SEP not in k}
+        lookup_parameters = set(kwargs.keys()) - set(params.keys())
```

Comment:
No need to compute this if `lookup_parameters_allowed` is `True`. I don't know the ORM code enough, so you can ignore my comment if it sounds like a premature optimization:

```py
if not lookup_parameters_allowed:
    lookup_parameters = set(kwargs.keys()) - set(params.keys())
    if lookup_parameters:
        # ...
```